lib/report-generator.js

```diff
@@ -0,0 +1,12 @@
+const { genJsonReport } = require('./report-types/json-report.js');
+const { genJunitReport } = require('./report-types/junit-report.js');
+
+exports.generateReports = (reports, vfiles) => {
+  reports.forEach((report) => {
+    if (report.endsWith('.json')) {
+      genJsonReport(report, vfiles);
+    } else if (report.endsWith('junit.xml')) {
```

Comment:
How about checking if the , to be consistent with the `.json` check and the description of the `--report` flag?

```suggestion
    } else if (report.endsWith('.junit.xml')) {
```

Reply:
The only reason I wasn't consistent with that is so that you can create a report that is just called "junit.xml". If we add the `.`, then we require junit.xml to be the extension, and not just the full name of the file.

Since junit.xml isn't an extension, it's just a convention, it didn't have the `.`, however json is the extension.

The reason I didn't limit it to the `.xml` ending, but instead added `junit` in front of it, is because you may choose to add additional XML report types in the future, and I didn't want to make it so that you'd have to implement a BC to add them.

Reply:
Got it! Thanks for the explanation.